torch/distributed/_tensor/sharding_prop.py

```diff
@@ -1,4 +1,5 @@
-from typing import Callable, Dict, Optional, Tuple
+from functools import lru_cache
```

Comment:
Does this one comes with cache missing rate stats? So that we can debug or see how the cache is working.

Reply:
added a API to explicitly get the missing rate/hit rate https://github.com/pytorch/pytorch/pull/107305/files#diff-bc0598e52e1f48ecf4f904ebbe5fd206854933871c3ffe305f95885ecbf53edcR4